File: njob-rest/src/main/java/com/github/nsorin/njobrest/model/Document.java
package com.github.nsorin.njobrest.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.validation.constraints.NotNull;

import lombok.Getter;
import lombok.Setter;

@Entity
public class Document {

    @GeneratedValue
    @Id
    @Getter
    @Setter
    private Integer id;

    @Getter
    @Setter
    @NotNull
    private String name;

    @Getter
    @Setter
    @NotNull
    private String fileName;

    @Getter
    @Setter
    @NotNull
    private String fileType;

    @Getter
    @Setter
    @NotNull
    private String fileKey;

    @ManyToOne
    @Getter
    @Setter
    private Application application;
}

File: njob-frontend/src/store/models/Document.js
import { Model } from "@vuex-orm/core"
import Application from "./Application"

export default class Document extends Model {

    static entity = 'document'

    static fields() {
        return {
            id: this.number(null),
            name: this.string(''),
            fileName: this.string(''),
            fileType: this.string(''),
            fileKey: this.string(''),
            application_id: this.number(null),
            application: this.belongsTo(Application, 'application_id')
        }
    }
}